mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -314,6 +315,21 @@ class Word2Vec extends Serializable with Logging {
     val expTable = sc.broadcast(createExpTable())
     val bcVocab = sc.broadcast(vocab)
     val bcVocabHash = sc.broadcast(vocabHash)
+    try {
+      doFit(dataset, sc, expTable, bcVocab, bcVocabHash)
+    }
+    finally
```

Comment:
I already spotted and corrected a few instances of forgotten broadcasted variables. I've already reported and patched all I was aware off.

I was suggesting introducing an ARM pattern and making broadcasted variable ARM aware (see comment above). This is by no way a general solution but it might help reduce the issue when the use of a broadcasted variable is compatibble with a syntactic scope. @thunterdb legitimately raised concern about introducing new dependencies (to scala-arm specifically). What is your insight wrt to this ?